model/generate_test.go

```diff
@@ -526,25 +526,40 @@ func (s *GenerateSuite) TestParseProjectFromJSON() {
 }
 
 func (s *GenerateSuite) TestValidateMaxVariants() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	p := &Project{}
 	g := GeneratedProject{}
-	s.NoError(g.validateMaxTasksAndVariants())
+	s.NoError(g.validateMaxTasksAndVariants(ctx, p))
 	for i := 0; i < maxGeneratedBuildVariants; i++ {
 		g.BuildVariants = append(g.BuildVariants, parserBV{})
 	}
-	s.NoError(g.validateMaxTasksAndVariants())
+	s.NoError(g.validateMaxTasksAndVariants(ctx, p))
 	g.BuildVariants = append(g.BuildVariants, parserBV{})
-	s.Error(g.validateMaxTasksAndVariants())
+	s.Error(g.validateMaxTasksAndVariants(ctx, p))
 }
 
 func (s *GenerateSuite) TestValidateMaxTasks() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	settings := &evergreen.Settings{
+		TaskLimits: evergreen.TaskLimitsConfig{
+			MaxTasksPerVersion: 25000,
+		},
+	}
+	s.NoError(evergreen.UpdateConfig(s.ctx, settings))
```

Comment:
Can we set this back to the original admin settings at the end of the test? The unit tests have an unfortunate problem where a lot of them assume that they're running with the initial admin settings set by `testutil.Setup()`, so modifying the DB admin settings initialized by `init` could mess with other unrelated tests.

Would be nice to fix one day, but I don't have great suggestions in the meantime without significantly overhautling the tests.